### DISCUSSION TITLE
How many vCPUs and Memory I need to operate k3s cluster which has 3 server nodes and 1.2TB HDD with cubeFS?

### DISCUSSION THREAD

**Discussion by Bokki-Ryu** [cubefs/cubefs#3436] [Q&A]
I plan to operate k3s cluster with cubeFS.

I don't know minimum hardware requirements for it.

Please give me advices to run cubeFS with k3s.

**Comment by leonrayang** [comment]
It is recommended to use more than 4 nodes(Three nodes can work well, but there is no disaster recovery.), with each node configured with at least 16 cores, 16GB of RAM, and 1TB of storage per disk.
